packages/backend/src/core/discovery/config/ConfigReader.ts

```diff
@@ -35,14 +35,14 @@ export class ConfigReader {
     return result
   }
 
-  async readDiscovery(name: string): Promise<ProjectParameters> {
+  async readDiscovery(name: string): Promise<DiscoveryOutput> {
     const contents = await readFile(
       `discovery/${name}/discovered.json`,
       'utf-8',
     )
 
     const parsed: unknown = JSON.parse(contents)
 
-    return parsed as ProjectParameters
+    return parsed as DiscoveryOutput
```

Comment:
maybe we should consider defining DiscoveryOutput in Zod? Then we would just do `DiscoveryOutput.parse(JSON.parse(contents))`

Reply:
I think it's out of scope of this PR but I'm going to add a task for it for the future